src/services/createRouter.ts

```diff
@@ -72,17 +70,17 @@ export function createRouter<
   const TRoutes extends Routes,
   const TOptions extends RouterOptions,
   const TPlugin extends RouterPlugin = EmptyRouterPlugin
->(arrayOfRoutes: TRoutes[], options?: TOptions, plugins?: TPlugin[]): Router<TRoutes, TOptions, TPlugin>
+>(routes: TRoutes[], options?: TOptions, plugins?: TPlugin[]): Router<TRoutes, TOptions, TPlugin>
 
 export function createRouter<
   const TRoutes extends Routes,
   const TOptions extends RouterOptions,
   const TPlugin extends RouterPlugin = EmptyRouterPlugin
 >(routesOrArrayOfRoutes: TRoutes | TRoutes[], options?: TOptions, plugins: TPlugin[] = []): Router<TRoutes, TOptions, TPlugin> {
-  const pluginRoutes = plugins.map((plugin) => plugin.routes)
-  const routes = insertBaseRoute([...routesOrArrayOfRoutes, ...pluginRoutes].flat(), options?.base)
+  const routes = getRoutesForRouter(routesOrArrayOfRoutes, plugins, options?.base)
+  const hooks = createRouterHooks()
 
-  checkDuplicateNames(routes)
```

Comment:
This is unrelated. But part of the pattern of "pass the router and plugin arguments into a utility to combine them and return a normalized response".